llvm/lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -4892,10 +4893,14 @@ Error BitcodeReader::parseFunctionBody(Function *F) {
         }
       } else {
         auto CastOp = (Instruction::CastOps)Opc;
+
         if (!CastInst::castIsValid(CastOp, Op, ResTy))
           return error("Invalid cast");
         I = CastInst::Create(CastOp, Op, ResTy);
       }
+      if (OpNum < Record.size() && isa<NonNegInstruction>(I) &&
+          (Record[OpNum] & (1 << bitc::NNI_NON_NEG)))
```

Comment:
nit: think `bitc:NNI_NON_NEG` can just be `1 << bitc::NNI_NON_NEG` (as opposed to the log). All the uses I see are as the mask and that seems to fit the pattern of other enums better as well.

Reply:
If there is a reason to keep as log then this should be `ResTypeID(1) << ...` and below should be `uint64_t(1) << ...`

Reply:
This is just following the same convention used for the other flags (OBO and PEO): https://github.com/llvm/llvm-project/blob/e93bddb2874db5520dfedecac6c6a13ffa4e7e5e/llvm/lib/Bitcode/Reader/BitcodeReader.cpp#L4849-L4863

Reply:
Okay to keep as log then (although I'd say preferably all would be changed to be the mask), but then can you cast `1` to the proper shift width. Otherwise just feels like bug waiting to happen.

Reply:
Is the "bug waiting to happen" that we are going to add 30 more flags to this instructions? That seems like a pretty contrived concern, especially when instructions can only have up to 7 flags.

Reply:
Fair enough